metabase/metabase
# feat(sdk): support multiple interactive questions by decoupling from query builder reducer

Closes https://github.com/metabase/metabase/issues/44670

### Description

Supports using multiple interactive questions at a time by storing the states in React context, as opposed to relying on the query builder states. This PR replaces our use of query builder actions and selectors (e.g. `initQBRaw`, `updateQuestion`, etc.) with logic implemented in the SDK.

### How to verify

- Go to the `test-multiple-instances-interactive-questions` branch in the demo app.
- Go to `/dev`, the kitchen sink page.
- You should be able to see multiple interactive questions and drill-down now. These interactive questions should be independent.

### Demo

<div>
    <a href="https://www.loom.com/share/10687eceba404ab59a416ce0d0bafce3">
      <p>Multiple Interactive Questions Demo - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/10687eceba404ab59a416ce0d0bafce3">
      <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/10687eceba404ab59a416ce0d0bafce3-with-play.gif">
    </a>
  </div>

### Checklist
